tests/e2e/builder.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// setupWorkSpace checks out given revision in a tmp directory
+// and returns path where the workspace is located
+func setupWorkSpace(revision string) (string, error) {
+	workSpace, err := os.MkdirTemp(os.TempDir(), "e2eWorkTree_")
+	if err != nil {
+		return "", fmt.Errorf("error creating temp directory %v", err)
+	}
+
+	fmt.Printf("Setting up worktree in '%s'\n", workSpace)
+
+	cmd := exec.Command("git", "worktree", "add",
+		"--checkout", workSpace, revision)
+	var errbuf bytes.Buffer
+	cmd.Stderr = &errbuf
+	fmt.Printf("Running: %s", cmd.String())
+	if err := cmd.Start(); err != nil {
```

Comment:
nit: this is just doing Start and then Wait, can probably be combined into one call, right? cmd.CombinedOutput, maybe?